### DISCUSSION TITLE
Rest API `User` deprecation

### DISCUSSION THREAD

**Discussion by ryandutton** [apache/airflow#36624] [General] [answered]
It looks like as of v2.8.0 the user endpoints are being deprecated, can be seen [here](https://airflow.apache.org/docs/apache-airflow/2.8.0/stable-rest-api-ref.html#tag/User). There doesn't seem to be any mention of this in the release notes. Is the deprecated intentional and it's been missed in the release notes _or_ is there a mistake with the deprecation notices. 

If this decision was intentional it would be good if you could direct me to any discussions or decisions around the deprecation of these endpoints as I make use of these and will have to find an alternative method to manage users.

**Comment by potiuk** [comment, marked as answer]
cc: @vincbeck - seems that users are reading our deprecation notices :) 

We moved the URL as we are preparing to separate `fab` to a provider and then it will move the API url to reflect that, however we have automated redirction from the old URL. I think we do need to add a bit more explanation on that (or maybe even remove the deprecation at all) - but generaly speaking you can largely ignore it for now @ryandutton . Even if we decide to remove it, it will not be removed until Airflow 3 is out (and we even do not have discussions whether there will ever be Airlfow 3).

**Reply by ryandutton** [reply]
Thank you for the swift response, it makes it all much clearer and I won't worry about it now.

**Reply by vincbeck** [reply]
We should do either one of these things:
- Keep the deprecation warning and encourage users to migrate and use the fab endpoint
- Remove the deprecation warning

Overall, I dont like asking users to ignore deprecation warning, it is not a good practise. My opinion is to keep the deprecation warnings and encourage users to use the fab endpoint (btw, the deprecation warning explicitly says "This API endpoint is deprecated. Please use the API under /auth/fab/v1 instead for this operation.". We might want to give more instructions/details) in case one day we decide to change the default auth manager (in Airflow 3 for instance). But I can feel you think otherwise @potiuk :)

**Reply by potiuk** [reply]
Nope. I am 100% with you on that one. 

But the thing is that the Swagger UI does NOT show the explanation. If we could fix THAT, then I am 100% fine with the deprecation staying as it is:

![image](https://github.com/apache/airflow/assets/595491/e8f68cf0-054c-4f9f-8d47-af9071a7ef75)

See https://airflow.apache.org/docs/apache-airflow/2.8.0/stable-rest-api-ref.html#tag/User

**Reply by vincbeck** [reply]
Sounds good! Let me do that then

**Reply by vincbeck** [reply]
#36664

**Reply by ryandutton** [reply]
I think if the reason for the deprecation was shown in the Swagger documentation was present this would have all been much clearer and no reason to raise the discussion point. Thank you both for actioning this.
